Cython/Compiler/Naming.py

```diff
@@ -135,6 +135,7 @@
 cython_runtime_cname   = pyrex_prefix + "cython_runtime"
 cyfunction_type_cname = pyrex_prefix + "CyFunctionType"
 fusedfunction_type_cname = pyrex_prefix + "FusedFunctionType"
+dataclass_field_default_cname = pyrex_prefix + "dataclass_dflt"
```

Comment:
I think this was because picked because the CPython implementation uses a variable called `f"_dflt_{field.name}"` to store the default values. https://github.com/python/cpython/blob/4868b94c6089d457673b1ba5b5b64c2f38c435af/Lib/dataclasses.py#L454. What I'm doing isn't *exactly* the same, but that's where I got it from

In both cases it doesn't really matter - it's a hidden variable that the user shouldn't really be able to get to so we can name it whatever we like. But I have a mild preference for using the short name "by analogy" to the CPython implementation.

For now I'll keep the short name and add an explanatory note. Change it if you disagree with my preference though!